docker-compose.yml

```diff
@@ -1,49 +1,102 @@
-# Docker Compose file Reference (https://docs.docker.com/compose/compose-file/)
-
 version: '3.4'
 
-# Define services
 services:
-#### Uncomment section bellow if you want to start the latest go-ipfs instance
-  cas:
-    # Configuration for building the docker image for the service
-    build:
-      context: . # Use an image built from the specified dockerfile in the current directory.
-      dockerfile: Dockerfile
-    restart: unless-stopped
-    environment: # Pass environment variables to the service
+
+  cas_api:
+    build: .
+    environment:
       - NODE_ENV=dev
       - APP_MODE=server
       - APP_PORT=8081
-      - ANCHOR_SCHEDULE_EXPRESSION=0 0 */1 * * *
-      - IPFS_API_URL=ipfs:5001
+
+      - ANCHOR_CONTROLLER_ENABLED=false
+      - ANCHOR_EXPIRATION_PERIOD=0
+      - ANCHOR_SCHEDULE_EXPRESSION=0/1 * * * ? *
+
       - BLOCKCHAIN_CONNECTOR=ethereum
-      - ETH_NETWORK=ganache
+      - CERAMIC_API_URL=http://host.docker.internal:7007
       - ETH_GAS_LIMIT=4712388
       - ETH_GAS_PRICE=100000000000
-      - ETH_RPC_HOST=http://ganache
-      - ETH_RPC_PORT=8545
-      - ETH_WALLET_PK=0x06dd0990d19001c57eeea6d32e8fdeee40d3945962caf18c18c3930baa5a6ec9
+      - ETH_NETWORK=ropsten
+      - ETH_OVERRIDE_GAS_CONFIG=false
+      - ETH_WALLET_PK=0x16dd0990d19001c50eeea6d32e8fdeef40d3945962caf18c18c3930baa5a6ec9
+
+      - IPFS_API_URL=ipfs:5001
+      - IPFS_API_TIMEOUT=10000
+
       - LOG_LEVEL=debug
       - LOG_TO_FILES=true
-      - TYPEORM_CONNECTION=sqlite
-      - TYPEORM_HOST=localhost
+      - LOG_PATH=/usr/local/var/log
+      - MERKLE_DEPTH_LIMIT=0
+
+      - TYPEORM_CONNECTION=postgres
+      - TYPEORM_DATABASE=anchor_db
+      - TYPEORM_HOST=database
+      - TYPEORM_PORT=5432
       - TYPEORM_USERNAME=root
       - TYPEORM_PASSWORD=root
-      - TYPEORM_DATABASE=./anchor.db.sq3
-      - TYPEORM_PORT=5432
+
+      - TYPEORM_ENTITIES=build/models/**/*.js
+      - TYPEORM_SUBSCRIBERS=build/subscribers/**/*.js
+      - TYPEORM_LOGGING=false
       - TYPEORM_SYNCHRONIZE=true
-      - TYPEORM_LOGGING=true
+
+      - VALIDATE_RECORDS=false
+    volumes:
+      - "/usr/local/var/log:/usr/local/var/log"
+    depends_on:
+      - ipfs
+    ports:
+      - "8081:8081"
+    networks:
+      - internal
+
+  cas_anchor:
+    build: .
+    restart: unless-stopped
```

Comment:
It seems like the API above should have `restart: unless-stopped`, but maybe `cas_anchor` shouldn't?

Reply:
What this achieves is basically forcing cas to keep restarting after the process exits

For the api we don't need this feature because if the process exits it is not intentional

The wording of this option is a bit misleading unfortunately. It is more accurately "unless explicitly stopped by docker stop command"

Reply:
> For the api we don't need this feature because if the process exits it is not intentional

Right, that's why we'd want docker to restart, to make sure the service stays up in the event of a crash, right?

For the anchor mode, we don't want it to restart exactly *because* we expect it to terminate in normal operation, and it should only be started up again by the scheduler job on a specific schedule, no?

Maybe I'm misunderstanding how this docker file fits into the rest of our infra management story

Reply:
No this docker-compose.yml is only used for testing locally. Not for testing infra.

Reply:
ah okay, then I guess it doesn't matter too much what the retry strategy here is.  Making the anchor process continually restart constantly is kind of weird behavior, but so is running it just once and never again, so I guess this is fine then